b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/idpSection/otherComponents/idpDetailsSections/settings.tsx

```diff
@@ -0,0 +1,143 @@
+            });
+        }
+
+        return errors;
+    };
+
+    const onDataSubmit = (response: IdentityProviderFederatedAuthenticator) => {
+        if (response) {
+            successTypeDialog(toaster, "Changes Saved Successfully", "Idp updated successfully.");
+            fetchData();
+        } else {
+            errorTypeDialog(toaster, "Error Occured", "Error occured while updating the Idp. Try again.");
+        }
+    };
+
+    const onUpdate = async (values: Record<string, string>) => {
+        setLoadingDisplay(LOADING_DISPLAY_BLOCK);
+        controllerDecodeUpdateFederatedAuthenticators(session, idpDetails.id, federatedAuthenticators, values)
+            .then((response) => onDataSubmit(response))
+            .finally(() => setLoadingDisplay(LOADING_DISPLAY_NONE));
+    };
+
+    return (
+        <div className={ styles.addUserMainDiv }>
+
+            <div>
+                {
+                    federatedAuthenticators
+                        ? (<Form
+                            onSubmit={ onUpdate }
+                            validate={ validate }
+
+                            render={ ({ handleSubmit, submitting, pristine, errors }) => (
+                                <FormSuite
+                                    layout="vertical"
+                                    className={ styles.addUserForm }
+                                    onSubmit={ () => handleSubmit() }
+                                    fluid>
+
+                                    { federatedAuthenticators.properties
+                                        ? (<SettingsFormSelection
+                                            federatedAuthenticators={ federatedAuthenticators.properties }
+                                            templateId={ idpDetails.templateId } />)
+                                        : null
+                                    }
+
+                                    <div className="buttons">
+                                        <FormSuite.Group>
+                                            <ButtonToolbar>
+                                                <Button
+                                                    className={ styles.addUserButton }
+                                                    size="lg"
+                                                    appearance="primary"
+                                                    type="submit"
+                                                    disabled={ submitting || pristine || !checkIfJSONisEmpty(errors) }>
+                                                    Update
+                                                </Button>
+                                            </ButtonToolbar>
+                                        </FormSuite.Group>
+                                    </div>
+                                </FormSuite>
+                            ) }
+                        />)
+                        : null
+                }
+
```

Comment:
do we need this newline?

Reply:
Fixed with https://github.com/wso2/samples-is/pull/416/commits/3e0450172639aa334cdc6716a7e8640db58efa19